indigo_app/static/javascript/indigo/dom.js

```diff
@@ -0,0 +1,269 @@
+      if (offset !== 0) {
+        return node.splitText(offset);
+      } else {
+        return node;
+      }
+    }
+
+    node = range.commonAncestorContainer;
+    if (node.nodeType != Node.ELEMENT_NODE) node = node.parentElement;
+
+    // remove foreign elements while working with the range
+    Indigo.dom.withoutForeignElements(node, function() {
+      if (range.startContainer.nodeType === Node.TEXT_NODE) {
+        // split the start and end text nodes so that the offsets fall on text node boundaries
+        start = split(range.startContainer, range.startOffset);
+      } else {
+        // first text node
+        start = document.createNodeIterator(range.startContainer, NodeFilter.SHOW_TEXT).nextNode();
+        if (!start) return;
+      }
+
+      if (range.endContainer.nodeType === Node.TEXT_NODE) {
+        end = split(range.endContainer, range.endOffset);
+      } else {
+        end = range.endContainer;
+      }
+
+      // gather all the text nodes between start and end
+      iterator = document.createNodeIterator(
+        range.commonAncestorContainer, NodeFilter.SHOW_TEXT,
+        function (n) {
+          // ignore text nodes in weird positions in tables
+          if (ignore[n.parentElement.tagName]) return NodeFilter.FILTER_SKIP;
+          return NodeFilter.FILTER_ACCEPT;
+        });
+
+      // advance until we're at the start node
+      node = iterator.nextNode();
+      while (node && node !== start) node = iterator.nextNode();
+
+      // gather text nodes
+      while (node) {
+        posn = node.compareDocumentPosition(end);
+        // stop if node isn't inside end, and doesn't come before end
+        if ((posn & Node.DOCUMENT_POSITION_CONTAINS) === 0 &&
+          (posn & Node.DOCUMENT_POSITION_FOLLOWING) === 0) break;
+
+        nodes.push(node);
+        node = iterator.nextNode();
+      }
+    });
+
+    // mark the gathered nodes
+    nodes.forEach(function(node) {
+      var mark = document.createElement(tagName || 'mark');
+      node.parentElement.insertBefore(mark, node);
+      mark.appendChild(node);
+      if (callback) callback(mark);
+    });
+  };
+
+  /**
+   * Tweaked version of toRange from https://github.com/tilgovi/dom-anchor-text-position
+   * so that we can fix a bug that arises when selecting to the end of a node.
+   */
+  Indigo.dom.textPositionToRange = function(root, selector) {
+    if (root === undefined) {
+      throw new Error('missing required parameter "root"');
+    }
+
+    var document = root.ownerDocument;
+    var range = document.createRange();
+    var iter = document.createNodeIterator(root, NodeFilter.SHOW_TEXT);
+
+    var start = selector.start || 0;
+    var end = selector.end || start;
+    var count = domSeek(iter, start);
+    var remainder = start - count;
+
+    if (iter.pointerBeforeReferenceNode) {
+      range.setStart(iter.referenceNode, remainder);
+    } else {
+      range.setStart(iter.nextNode(), remainder);
```

Comment:
`iter.nextNode()` should be an object? 
`TypeError: Argument 1 of Range.setStart is not an object.`

Reply:
This is third party code. Do you know how you triggered this error?

Reply:
oh dear. In the example screenshot from above, I moved the subs (2) down.